### PR TITLE
fix: dependencies conflict

### PR DESCRIPTION
**Pull request by d4rkk3y**
address build fail after revanced-patcher be bump to new version.

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -67,6 +67,14 @@ flutter {
...
 }
 
+configurations.all {
+    resolutionStrategy {
+        dependencySubstitution {
+            substitute module('org.smali:dexlib2:2.5.2') using module('app.revanced:dexlib2:2.5.3-8649b642')
+        }
+    }
+}
+
```
Comment (oSumAtrIX) [inline android/app/build.gradle:77]: Why is this required?

Comment (d4rkk3y) [reply android/app/build.gradle:77]: because revanced-patcher depend on smali, multidexlib2 and apktool-lib. All of them depend on dexlib2, but multidexlib2 depend on different dexlib2(app.revanced:dexlib2:2.5.3-8649b642), while the smali and apktool-lib depend on org.smali:dexlib2:2.5.2. That make dependencies conflict and cause "duplicate class" issue.

Comment (Aunali321) [reply android/app/build.gradle:77]: LGTM

Comment (oSumAtrIX) [reply android/app/build.gradle:77]: In that case, this issue has to be fixed on the side of smali and apktool-lib.

Comment (d4rkk3y) [reply android/app/build.gradle:77]: > In that case, this issue has to be fixed on the side of smali and apktool-lib.

Yes, this is just a fix in case you don't want to maintain a addition apktool repo in revanced. So it is up to you.

Comment (oSumAtrIX) [reply android/app/build.gradle:77]: We would have to technically reflect that change on the CLI and any other application that uses the revanced-patcher dependency. For that reason, it must be fixed on the patcher side or its dependencies.

Comment (oSumAtrIX) [reply android/app/build.gradle:77]: `apktool-lib` depends on `smali` of which we already [have a fork](https://github.com/revanced/smali), as `dexlib2` is a submodule of it.

Comment (d4rkk3y) [reply android/app/build.gradle:77]: I see. So I guess it is better to fork apktool and bump dexlib2 in that repo.

Comment (oSumAtrIX) [reply android/app/build.gradle:77]: I was unable to find a dependency to dexlib2 with [this search](https://github.com/revanced/Apktool/search?q=dexlib) in our forked repository of `Apktool`.

Comment (d4rkk3y) [reply android/app/build.gradle:77]: > `apktool-lib` depends on `smali` of which we already [have a fork](https://github.com/revanced/smali), as `dexlib2` is a submodule of it, though `smali` uses our `dexlib2` fork as a dependency, the same as our `multidexlib2` fork:
> 
> https://github.com/revanced/smali/blob/master/smali/build.gradle#L75-L86.
> 
> Can you please elaborate on what we are missing?

I think we need fork apktool and make it use our smali and baksmali

Comment (oSumAtrIX) [reply android/app/build.gradle:77]: We already [have a fork](https://github.com/revanced/Apktool) which we already use.

Comment (d4rkk3y) [reply android/app/build.gradle:77]: https://github.com/iBotPeaches/Apktool/blob/a73ace23ff6872ecd2916a80623965d8cb1ed406/brut.apktool/apktool-lib/build.gradle#L40-L41 here

Comment (oSumAtrIX) [reply android/app/build.gradle:77]: In that case we just have to change [this](https://github.com/revanced/Apktool/blob/master/build.gradle#L21) and [that](https://github.com/revanced/Apktool/blob/master/build.gradle#L30) dependency, could you open a PR?

Comment (d4rkk3y) [reply android/app/build.gradle:77]: > In that case we just have to change [this](https://github.com/revanced/Apktool/blob/master/build.gradle#L21) and [that](https://github.com/revanced/Apktool/blob/master/build.gradle#L30) dependency, could you open a PR?

Sure. But I think you need release smali and baksmali package first.

Comment (oSumAtrIX) [reply android/app/build.gradle:77]: I was wondering since we do not need a fork of `smali` and `baksmali`, why are we releasing packages for it? We can simply depend on the upstream packages. Since our fork of `dexlib2` depends on it's own `smali` and `baksmali` modules,simply changing it to upstream would solve duplicates.

Comment (d4rkk3y) [reply android/app/build.gradle:77]: the problem is upstream `smali` and `baksmali` use dexlib2:2.5.2.

Comment (d4rkk3y) [reply android/app/build.gradle:77]: > Since our fork of `dexlib2` depends on its own `smali` and `baksmali` modules, simply changing it to upstream would solve duplicates.

I think smali and baksmali depend on dexlib2 not vice versa.

Comment (oSumAtrIX) [reply android/app/build.gradle:77]: I just noticed, that dexlib2 does not depend on `smali` nor `baksmali` but `apktool-lib`. Basically the dependency graph is the following:

```mermaid
graph TD;
    manager[ReVanced Manager]-->patcher;
    cli[ReVanced CLI]-->patcher;
    patcher[ReVanced Patcher]--> multidexlib2;
    patcher[ReVanced Patcher]--> apktool;
    multidexlib2[ReVanced Multidexlib2 fork]-->dexlib2fork;
    apktool[ReVanced Apktool fork]-->dexlib2fork[ReVanced Dexlib2 fork];
    apktool[ReVanced Apktool fork]-->smali;
    apktool[ReVanced Apktool fork]-->baksmali;
    smali[iBotPeaches Smali]-->dexlib2[iBotPeaches  Dexlib2];
    baksmali[iBotPeaches BakSmali]-->dexlib2[iBotPeaches Dexlib2];
```
